contrib/dracut/90zfs/module-setup.sh.in

```diff
@@ -5,8 +5,7 @@ check() {
 	[ "${1}" = "-d" ] && return 0
 
 	# Verify the zfs tool chain
-	which zpool >/dev/null 2>&1 || return 1
-	which zfs >/dev/null 2>&1 || return 1
+	require_binaries zpool zfs grep mount.zfs hostid awk head
```

Comment:
This will fail in old systems where require_binaries is not defined.  I believe all of the CentOS systems will simply fail, and that is at a very minimum.
Add more lines that use "which" instead, or top-level-define a compat require_binaries shell function (based on the one defined in your system) if one is undefined.